docs/LocalFileMetadataSource-batchsource.md

```diff
@@ -0,0 +1,41 @@
+| **Max Split Size**     |  **Y**   | None      | Specifies the number of files that are controlled by each split. The number of splits created will be the total number of files divided by Max Split Size. The InputFormat will assign roughly the same number of bytes to each split. |
+| **Copy Recursively**   |  **Y**   | True      | Whether or not to copy recursively. Similar to the `-r` option in the `cp` terminal command. Set this to true if you want to copy the entire directory recursively.                                                                    |
+
+Usage Notes
+-----------
+This source plugin only reads filemetadata from a local source filesystem. To copy files, pass its outputs to a FileCopySink of the destination Filesystem.
+A StructuredRecord with the following schema is emitted for each file it reads.
+
+| Field                  | Type   | Description                                                                                                                                    |
+| :--------------------- | :----- | :-------------------------                                                                                                                     |
+| **fileName**           | String | Only contains the name of the file.                                                                                                            |
+| **fullPath**           | String | Contains the full path of the file in the source file system.                                                                                  |
+| **fileSize**           | long   | File Szie in bytes.                                                                                                                            |
+| **hostURI**            | String | The source filesystem's URI.                                                                                                                   |
+| **modificationTime**   | long   | The modification timestamp of the file.                                                                                                        |
+| **group**              | String | The group that the file belongs to.                                                                                                            |
+| **owner**              | String | The owner of the file.                                                                                                                         |
+| **isFolder**           | Boolean| Whether or not the file is a folder.                                                                                                           |
+| **relativePath**       | String | The relative path is constructed by deleting the portion of the source path that comes before the last path separator ("/") from the full path.|
```

Comment:
Here's an example:
Say you do `cp -r  /foo/bar   /somefolder/` And there's a file `/foo/bar/baz/bak.txt`
The sourcePath in this case would be `/foo/bar`.
The full path of the file is `/foo/bar/baz/bak.txt`
The filename is only `bak.txt`
The relative path, constructed by dropping the prefix, would be `bar/baz/bak.txt`